### ISSUE TITLE
Add `--profile` to `kubectl run`

### ISSUE THREAD

**Issue by lindhe** [kubernetes/kubectl#1608]
**What would you like to be added**:

The `kubectl debug` command has a `--profile` flag to easily comply with [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/):

```console
$ kubectl debug --help
[…]
Options:
    […]
    --profile='legacy':
        Options are "legacy", "general", "baseline", "netadmin", "restricted" or "sysadmin".
```

It's an _amazing_ flag! I love it. It makes life so much easier when I'm trying to get a debug pod running quickly in a namespace that enforces the restricted policy.

I wish that `kubectl run` had a similar flag. Can we have that, pretty please? 🙏 

**Why is this needed**:

To make it easier for users to generate a pod that complies with a profile/PSS.

**Comment by ardaguclu** [comment]
Although that sounds a useful feature, I don't think we would want to add any new flag in `run` command and it is likely that this feature request will be rejected.

**Comment by lindhe** [comment]
Alright. Is `run` considered deprecated?

**Comment by ardaguclu** [comment]
> Alright. Is `run` considered deprecated?

We can't deprecate it as lots of scripts rely on this command. Instead, it is considered that no more flags will be added into this command.

**Comment by lindhe** [comment]
I see. Thanks for the info.

**Comment by lindhe** [comment]
Do you think I will have better luck proposing the same flag for `kubectl create deployment`?

**Comment by ardaguclu** [comment]
> Do you think I will have better luck proposing the same flag for `kubectl create deployment`?

I don't think so but let's keep this issue open and get a feedback regarding the `create deployment`

**Comment by ardaguclu** [comment]
I think, we have no plans to add new flags for `create deployment` as well too. I'd prefer closing this. Thank you.

/close
